core/src/main/java/org/elasticsearch/index/query/GeoShapeQueryParser.java

```diff
@@ -53,23 +47,22 @@
     }
 
     @Override
-    public Query parse(QueryShardContext context) throws IOException, QueryParsingException {
-        QueryParseContext parseContext = context.parseContext();
+    public GeoShapeQueryBuilder fromXContent(QueryParseContext parseContext) throws IOException, QueryParsingException {
         XContentParser parser = parseContext.parser();
 
         String fieldName = null;
-        ShapeRelation shapeRelation = ShapeRelation.INTERSECTS;
-        String strategyName = null;
-        ShapeBuilder shape = null;
+        ShapeRelation shapeRelation = null;
+        SpatialStrategy strategy = null;
+        BytesReference shape = null;
 
         String id = null;
         String type = null;
-        String index = DEFAULTS.INDEX_NAME;
-        String shapePath = DEFAULTS.SHAPE_FIELD_NAME;
+        String index = null;
+        String shapePath = null;
 
         XContentParser.Token token;
         String currentFieldName = null;
-        float boost = 1f;
+        Float boost = null;
```

Comment:
Basically so that the QueryBuilder can 'own' the default values. The parser then only sets what it collects from the request on the builder (checks if not null) and that way only the QueryBuilder has to worry about the default values.